### PR TITLE
Extend the immutable interface and cache allFields in rankprofile

### PR DESCRIPTION
**Pull request by baldersheim**
@bjorncs PR
@bratseth FYI
Brings perfomance further down from 15s to 13s

### CODE REVIEW

**Review by bratseth — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java [hunk 1]
```diff
@@ -15,11 +19,22 @@
  */
 public interface ImmutableSearch {
 
+    String getName();
+    Index getIndex(String name);
+    SDField getConcreteField(String name);
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java]: Consider changing the return type to `ImmutableSDField`

File: config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java [hunk 2]
```diff
@@ -15,11 +19,22 @@
...
+    List<SDField> allConcreteFields();
```
Comment (bjorncs) [inline config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java]: Consider changing the return type to `List<ImmutableSDField>`

**Review by bjorncs — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java (hunk 1 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java]: Will do in a follow up PR

File: config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java (hunk 2 above)
Comment (baldersheim) [reply config-model/src/main/java/com/yahoo/searchdefinition/ImmutableSearch.java]: Will do in a follow up PR

**Pull request merged**